[15.0][MIG] web_responsive: Migration to 15.0

SplashS opened the pull request:
Migration work has not been completed yet
But I share my work to get feedback and show cases to use OWL in v15

lasley committed:
[FIX] web_responsive: Other minor fixes
* Add Tecnativa as author
* Remove untestable JS method
* Rename to web_responsive

lasley committed:
[MIG] web_responsive: Migrate to v10
* Bump version
* Change all openerp to odoo
* Fix qunit suite injection
* Remove jQuery require
* Change to new selectors:
  * `oe_leftbar` to `o_sub_menu`
  * `oe_logo` to `o_sub_menu_logo`
  * `oe_footer` to `o_sub_menu_footer`
  * `oe_secondary_menus_container` to `o_sub_menu_content`
* Add style to hide oe_footer instead of remove to not break `support_branding`
* Add note in ReadMe explaining override of `support_branding`
* Set top margin of app drawer title to 0 to fit v10 proportions
* Fix notification badge positioning
* Add o_web_client class to body to fix overlaps
* Scroll control panel with page
* Change navbar z-index to not be overlapped by buttons
* Raise z-index on header to raise over buttons
* Handle layered notifications via z-index
* Remove `#` from navigation links HREF to prevent history littering

lasley committed:
[FIX] web_responsive: Change z-indexes
* Rebalance z-indexes to fix overlay issue while still providing notification support

oca-transbot committed:
Update translation files

danieltorres7 committed:
Form view improvements to fit smaller screens

On screens with less than 768px content is limited to 2 columns.

On mobile field labels have their own line.

Fix elements width for XS screens

danieltorres7 committed:
Form image editing controls always present

Make image editing controls always available, instead of depending
on resolution or hover.

danieltorres7 committed:
Web responsive menu improvements (#513)

* Bold titles for apps in the app-drawer
* Remove z-index from Control Panel buttons, so it doesn't overlap the menu.
* Better responsive menus
  * Move systray icons/menus out of the menu and up to the top bar.
  * Increase avatar size and visibility.

* Other minor menu behavior fixes.
* Changes to fit odoo variables
* And a little margin adjust on navbar-right
* Top menu always thick
* Removed overflow:hidden!important from main.less
* Local patch: prevent body overflow auto
* Stop using `style=` on `<body>`
* Fix jquery drawer in anticipation of:
   https://github.com/blivesta/drawer/pull/36

yajo committed:
Fix problem with overflowing chatter widget.

yajo committed:
[FIX][web_responsive] Form padding, titles and buttons OK in XS

Small makeup to make it look good in XS screens.

sergio-teruel committed:
[10.0][IMP] web_responsive: Load default kanban view for small screen

Commit:
[10.0][IMP][web_responsive] (#569)

* [IMP] drawer-toggle

* [IMP] Hitbox of close drawer-toggle

* [IMP] Removed outline on menu item

* [IMP] Logo positionings

* [FIX] Try me on runbot button

* [REM] Reversed outline change

zeroheure committed:
Update README to fix wikipedia link

lasley committed:
[FIX] web_responsive: Change edit accesskey
* Change accesskey for `edit` in form view back to `e` to fix #587

yajo committed:
[FIX][web_responsive] Make it compatible with l10n_es_toponyms

Before this patch, there was an incompatibility between this addon and l10n_es_toponyms, caused by the lack of wrapping of the special address field that was being added.

Now layout works as expected in that case. In any other case, it seems sensible to wrap inputs so no more weird overflows happen and less chances of needing horizontal scrolling happen.

pedrobaeza committed:
[FIX] web_responsive: Execution permission

Commit:
[10.0][IMP][web_responsive] Make search bar sticky  (#633)

* Initial commit

* Initial commit

yajo committed:
[FIX][web_responsive] Long-named multicompanies nice in XS (#688)

lasley committed:
[FIX] web_responsive: Chrome Slow Scrolling
* Update to iScoll bleeding edge to fix #741

angelmoya committed:
UPDT version

angelmoya committed:
FIX travis (#747)

* FIX hasclass

* FIX views

* [FIX] web_responsive: Syntax error in xpath

pedrobaeza committed:
[FIX] web_responsive: Correct runbot link

hgminh95 committed:
[MIG] web_responsive test to 11.0

lasley committed:
[FIX] web_responsive: Menu Height on Init
* Add a call to the window resize method during init to fix OCA#773

fanha99 committed:
[FIX] web_responsive: Hide header when print directly from browser issues 808 (#809)

Closes #808

lasley committed:
[IMP] web_responsive: Add search feature (#838)

* Add feature to search menus in the app drawer
* Switch model to RPC

Tardo committed:
[IMP] web_responsive: Possibility of putting the chatter in the right  (#951)

OCA-git-bot committed:
[UPD] README.rst

Tardo committed:
[FIX] Chatter Position

Tardo committed:
[FIX] Read/Write permissions

Tardo committed:
[CLEAN] Comments

Tardo committed:
[IMP] Code

Tardo committed:
[IMP] Adding res.users chatter_position read/write test

yajo committed:
Fix linter

Tardo committed:
[IMP] Sticky Statusbar

Tardo committed:
[IMP] Add sticky header & footer in list view

Tardo committed:
[FIX][11.0][web_responsive] Views without sheet

yajo committed:
[FIX] web_responsive: Complete search feature forward-porting

Tardo committed:
[IMP][11.0][web_responsive] Suggested changes (#978)

* [IMP] Chatter Position
* [IMP] Clean unused stuff
* [IMP] FieldStatus hidden-xs
* [IMP] Improvements suggested in #540
* [WIP] Suggested changes
* [WIP] Improve FieldStatus-Responsive
* [IMP] Statusbar Buttons & FieldStatus
* [IMP] Statusbar Buttons Style
* [IMP] Button 'Task' style

yajo committed:
[FIX] web_responsive: Disable access to hidden menus

When doing menu search, some menus that the user didn't have permission to use were presented to him.

Fix #850.

pedrobaeza committed:
[IMP] web_responsive: Add known issue note about sticky

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

yajo committed:
[MIG] web_responsive: Migrate to v12 and refactor

This migration includes a full refactoring to make this module more
maintainable. Some things that have changed:

- Removed external libraries.
- Change Less for Scss.
- Reduce ES and XML to the minimal required needs.
- Implement as much features as possible with just Scss.
- Remove copyright from `__init__.py` files.
- Trigger the new hotkeys system from Odoo v12 with `Shift+Alt` instead
  of just `Alt`, and restore some good old hotkeys (`E` for "Edit",
  `D` for "Discard", and `A` for "Apps menu").
  See https://github.com/odoo/odoo/issues/30068 on the matter.
- Control panel breadcrumbs are collapsed into a single backwards icon.
- Add FA icons to most common buttons in control panel.
- Hide text in XS for those buttons, to have a slicker phone experience.
- Lots of gifs in the README!

oca-travis committed:
Update translation files

yajo committed:
[FIX] web_responsive: Skip re-search when not writing

When the user uses TAB or SHIFT+TAB to navigate in search results, he can trigger a SHIFT-only keydown event without noticing, which would reset the selected result position to the first or last.

It is disturbing, so, to avoid that problem, the re-search is only triggered if the currently pressed key has length=1 or is Backspace, which will skip most keys that are not actually writing a character into the search input, like i.e. "Shift", "Alt", "F4", etc.

In addition to that, to detect if the search results is empty, the `:empty` selector is not trustable because it considers not empty nodes with whitespace-only content. That has been patched too.

yajo committed:
[FIX] web_responsive: Make app icons and search results middle-button-clickable

In the case of app icons, those weren't getting the `href` attribute filled.

In the case of search results, those weren't even an `<a>` element.

Fixed in both cases to fix https://github.com/OCA/web/issues/1170.

Tardo committed:
[IMP] Sticky chatter topbar

Tardo committed:
[FIX+IMP] web_responsive: Chatter Topbar + fix #1189 + added gif

yajo committed:
Update web_responsive/static/src/css/web_responsive.scss

Co-Authored-By: Tardo <dev@redneboa.es>

yajo committed:
[FIX] web_responsive: Allow searching apps in mobile Chrome

This patch fixes #1251.

The keydown event is faked in mobile chrome because it's not something reliable due to today's virtual keyboards prediction features. Typing a <kbd>C</kbd> character doesn't mean it will be there when user uses autocorrection/autoguessing.

Instead of that, we use now the `keydown` event exclusively for movement features, and `input` event exclusively for content changes in the search input element.

Apart from fixing search in mobile Chrome, it also makes code more readable.

The search input also disables autocompletion since it would be useless and annoying here.

Tardo committed:
[IMP] web_responsive: chatter width

Tardo committed:
[IMP] web_responsive: Sticky in tree view and form view

Tardo committed:
[IMP] web_responsive: Support long title in form view

Tardo committed:
[IMP] web_responsive: AppMenu wait for load view

Tardo committed:
[IMP] web_responsive: Suggested changes

Tardo committed:
[IMP] web_responsive: Menu wait for action

Tardo committed:
[IMP] web_responsive: Readme

Tardo committed:
[FIX] web_responsive: AppDrawer in Chromium

Tardo committed:
[IMP] web_responsive: Refactor code

Tardo committed:
[IMP] web_responsive: Readme

Tardo committed:
[FIX] web_responsive: Chatter position

Tardo committed:
imp overflow

Tardo committed:
[IMP] web_responsive: Chatter on small screen

Tardo committed:
[FIX] web_responsive: Close AppDrawer

keshrath committed:
[FIX] web_responsive: AppDrawer Discard

keshrath committed:
[FIX] web_responsive: adapt AppDrawer Discard

keshrath committed:
[FIX] web_responsive: Close Menu on Discard

OCA-git-bot committed:
[UPD] README.rst

keshrath committed:
Update web_responsive.scss

Tardo committed:
[IMP] web_responsive: Performance when hide menus

Tardo committed:
[IMP] web_responsive: Don't show menu if waiting for an action to end

Tardo committed:
[IMP] web_responsive: Small refactor 'isWaiting'

Tardo committed:
[FIX] web_responsive: changed button box more initial position

Tardo committed:
[MIG] web_responsive: Migration to 13.0

Tardo committed:
[IMP] web_responsive: Made chatter input & attachments scrollable

Tardo committed:
[IMP] web_responsive: Support long titles

Tardo committed:
black, isort

pedrobaeza committed:
[IMP] web_responsive: Add development status + contributors

OCA-git-bot committed:
[UPD] README.rst

emagdalenaC2i committed:
[FIX] web_dialog_size js warnings

Tardo committed:
[IMP] web_responsive: Adapt test to 13.0

Tardo committed:
[IMP] mail_tracking: Changed form statusbar z-index

Tardo committed:
[IMP] web-responsive: Document Viewer

Commit:
[FIX] web_responsive: make followers dropdown usable

Fix a visual glitch that impeded using the followers dropdown with a sided chatter.

Commit:
[IMP] pre-commit run after update

Includes some manual fixes to silent ESLint warnings.

Tardo committed:
[FIX] web_responsive: Change AppDrawer close method

Previous of this commit, AppDrawer closes when the action have an id
that is defined in a menu item and no rpc calls are running.
Now AppDrawer closes in a more generic way, is enough that a
controller is attached to consider that the AppDrawer needs to be
closed.

sbidoul committed:
pre-commit update: prettier xml indent

kos94ok-3D committed:
[FIX] web_responsive: fix hotkey combinations in a shortcut template

Tardo committed:
[IMP] web_responsive: Restore discuss mobile mode

androettop committed:
New style in apps menu

scss file to change ui variables easily

Deleted unnecesary padding

remove unnecesary styles

unnecesary styles

align menu items to left again

search results styles and fixes

prettier configuration fix

newline at end of file (variables.scss)

text color calculation for navbar

fix in img url

requested fixes

Transitions and focus fixes

fix in js settimeout

margin added in search-container

Fix in transition

Co-authored-by: Jairo Llopis <Yajo@users.noreply.github.com>

Remove duplicated css

ivantodorovich committed:
[IMP] Improve responsiveness in large screens: don't let the icons separate too much from each other

[FIX] Display of long search result items in mobile

[IMP] AppMenu searchbar style

[IMP] Add vertical padding to icons, to increase separation

[IMP] Add focus style on app icons

[IMP] Prevent user selection on menu

[IMP] Prevent dragging the menues

OCA-git-bot committed:
[UPD] README.rst

pedrobaeza committed:
[IMP] web_responsive: Adjust roadmap

* Removed one of the wishlist after #1654 and #1657
* Added the bug depicted in #1626

Closes #1626

OCA-git-bot committed:
[UPD] README.rst

Tardo committed:
[FIX] web_responsive: Hide search panel on small screens

Tardo committed:
[IMP] web_responsive: Cut long filters names in the filters menu

Tardo committed:
[IMP] web_responsive: Apply pre-commit changes

Tardo committed:
[FIX] web_responsive: Menu section overflow

Tardo committed:
[IMP] web_responsive: Bigger checkboxes in list view

ypapouin committed:
[FIX] web_responsive: Do not set overflow property on inline fields
Co-authored-by: Alexandre Díaz <alexandre.diaz@tecnativa.com>
Ensure that the title is displayed while the user holds the mouse click (while selecting content)

Commit:
[MIG] web_responsive: Migration to 14.0

Tardo committed:
[IMP] web_responsive: Scrollable dropdowns

oca-travis committed:
[UPD] Update web_responsive.pot

OCA-git-bot committed:
[UPD] README.rst

SplashS committed:
[FIX] error on grouped mobile kanban with zero groups

SplashS committed:
Added translation using Weblate (Russian)

SplashS committed:
Translated using Weblate (Russian)

Currently translated at 87.5% (21 of 24 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/ru/

Yvesldff committed:
Added translation using Weblate (French (France))

Yvesldff committed:
Translated using Weblate (French (France))

Currently translated at 50.0% (12 of 24 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fr_FR/

SplashS committed:
[FIX] word wrap for appdrawer menu items

bosd committed:
Translated using Weblate (Dutch)

Currently translated at 77.7% (14 of 18 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/nl/

remi-filament committed:
Translated using Weblate (French)

Currently translated at 77.7% (14 of 18 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fr/

saeed-raeisi committed:
Added translation using Weblate (Persian)

saeed-raeisi committed:
Translated using Weblate (Persian)

Currently translated at 100.0% (24 of 24 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fa/

saeed-raeisi committed:
Added translation using Weblate (Persian)

remi-filament committed:
Translated using Weblate (French)

Currently translated at 88.8% (16 of 18 strings)

Translation: web-14.0/web-14.0-web_responsive
Translate-URL: https://translation.odoo-community.org/projects/web-14-0/web-14-0-web_responsive/fr/

SplashS committed:
[MIG] web_responsive: Migration to 15.0

pedrobaeza commented:
Thank you very much!

SplashS commented:
So I think I stabilized migration.
I have extra plan only to recapture screenshots. They are definitely outdated.

@pedrobaeza Some notes about hotkey overlays:

1. I left Alt modifier as in base odoo, but I added Alt + Shift modifier only for navbar menu entries.
This variant excludes conflict with Firefox tab switching. All current hotkeys works well in Chrome and Safari.
2. I tried to change Prev page and Next page hotkeys to Z and X accordingly. I think we should discuss it.

Also I made correct handling of window resize. Interface changed dynamically on device rotation f.x.

There are two implementations of ControlPanel. I was forced to patch both with code duplication. I wish in 16.0 there will be only one.

Waiting for feedback.

SplashS committed:
[IMP] show AppsMenu by default after login
also redirect to backend from website AppsMenu

SplashS commented:
@Tardo Migration is almost ready and runbot build named `runboat/build` works well.
This PR is ready for review.

Tardo commented on web_responsive/static/src/components/navbar/main_navbar.esm.js:
```diff
@@ -0,0 +1,13 @@
+};
+registry
+    .category("systray")
+    .add("web.user_menu", systrayItem, {force: true, sequence: 0});
```
Why is this necessary?

SplashS replied:
It was necessary before https://github.com/odoo/odoo/commit/3b18f1cf8cfdbc497c8acba67970e835dabc2eb2#diff-c8d085006528c9e527b7d7575894d1ab86722180c60be7a75f50187ab6c5d14f
Now it is unnecessary. I will remove it

Thanks for feedback! I will consider it soon and let you now

Tardo reviewed (commented):
Thanks, good job! I have seen some little "glitches":
- Tree Kanban doesn't look as in previous versions:
![responsive_15_kanban_tree](https://user-images.githubusercontent.com/731270/141521628-0bed4584-dcf5-46c9-bdd6-401004df585e.png)

- The attachment zone of the chatter should be fixed when use the scroll:
![responsive_15_attachments_zone](https://user-images.githubusercontent.com/731270/141521737-e2304048-7f55-4e1a-8d0c-bfc32cdc08eb.png)

It may also come in handy to refactor to change the use of "var" to let or const, avoid the use of "self" by using "arrow functions", etc..

pedrobaeza commented:
/ocabot migration web_responsive

SplashS committed:
[IMP] recaptured actual description screenshots

SplashS commented:
@Tardo 
> * Tree Kanban doesn't look as in previous versions:

Fixed

> * The attachment zone of the chatter should be fixed when use the scroll:

I looked at 14.0 version. Attachments zone isn't fixed. In my opinion it's dangerous to make it fixed because attachments count isn't limited. 

> It may also come in handy to refactor to change the use of "var" to let or const, avoid the use of "self" by using "arrow functions", etc..

Done

Tardo committed:
[IMP] web_responsive: Adapt label width

SplashS committed:
[IMP] big inputs on forms in edit mode

Tardo committed:
[FIX] web_responsive: Label width on modal

jcdrubay commented:
Tested in local and I could not see the sub menus

I could not test on runbot nor runboat, maybe a new build needs to be triggered by a new commit ?

![image](https://user-images.githubusercontent.com/1765420/147333817-1fe42046-4b60-44e7-bc4e-702b42bac0eb.png)

SplashS commented:
> Tested in local and I could not see the sub menus
> 
> I could not test on runbot nor runboat, maybe a new build needs to be triggered by a new commit ?


Thank you very much for feedback.
It's broken since https://github.com/odoo/odoo/commit/3659d65f37d08737f99d62ecb8cd486ba8d764f4
Fixed.

Runboat will work few days after my push.

jcdrubay commented:
Thanks @SplashS 
Runboat is now working well and I could see the sub-menus nicely 👍

jcdrubay reviewed (approved):
LGTM after running few tests on Runboat. Thanks @SplashS

stfnndrsn commented:
I just tried to install `web_responsive` on a clean v15  database; but I get the following exception

```RPC_ERROR
Odoo Server Error
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/tools/cache.py", line 85, in lookup
    r = d[key]
  File "/usr/lib/python3/dist-packages/odoo/tools/func.py", line 71, in wrapper
    return func(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/tools/lru.py", line 34, in __getitem__
    a = self.d[obj]
KeyError: ('ir.model.data', <function IrModelData._xmlid_lookup at 0x7f2c710e8dc0>, 'web.assets_backend')

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 680, in _tag_root
    f(rec)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 658, in _tag_template
    return self._tag_record(record)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 567, in _tag_record
    f_val = self.id_get(f_ref)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 663, in id_get
    res = self.model_id_get(id_str, raise_if_not_found)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 669, in model_id_get
    return self.env['ir.model.data']._xmlid_to_res_model_res_id(id_str, raise_if_not_found=raise_if_not_found)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_model.py", line 1935, in _xmlid_to_res_model_res_id
    return self._xmlid_lookup(xmlid)[1:3]
  File "<decorator-gen-35>", line 2, in _xmlid_lookup
  File "/usr/lib/python3/dist-packages/odoo/tools/cache.py", line 90, in lookup
    value = d[key] = self.method(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_model.py", line 1928, in _xmlid_lookup
    raise ValueError('External ID not found in the system: %s' % xmlid)
ValueError: External ID not found in the system: web.assets_backend

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_http.py", line 237, in _dispatch
    result = request.dispatch()
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 688, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 360, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 94, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 349, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 917, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 536, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 1339, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 1327, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 464, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 451, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-74>", line 2, in button_immediate_install
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_module.py", line 74, in check_and_log
    return method(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_module.py", line 483, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_module.py", line 600, in _button_immediate_function
    registry = modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/usr/lib/python3/dist-packages/odoo/modules/registry.py", line 87, in new
    odoo.modules.load_modules(registry, force_demo, status, update_module)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 474, in load_modules
    processed_modules += load_marked_modules(cr, graph,
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 363, in load_marked_modules
    loaded, processed = load_module_graph(
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 222, in load_module_graph
    load_data(cr, idref, mode, kind='data', package=package)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 69, in load_data
    tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 745, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 811, in convert_xml_import
    obj.parse(doc.getroot())
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 731, in parse
    self._tag_root(de)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 693, in _tag_root
    raise ParseError('while parsing %s:%s, somewhere inside\n%s' % (
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 644, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 302, in _handle_exception
    raise exception.with_traceback(None) from new_cause
odoo.tools.convert.ParseError: while parsing None:12, somewhere inside
<data name="Open Mobile Assets" inherit_id="web.assets_backend">
        <xpath expr=".">
            <link rel="stylesheet" href="/web_responsive/static/src/css/web_responsive.scss"/>
            <script type="application/javascript" src="/web_responsive/static/src/js/web_responsive.js"/>
            <script type="application/javascript" src="/web_responsive/static/src/js/discuss.js"/>
        </xpath>
    </data>
```

Tardo reviewed (changes requested):
Overall it is a great job ;)

I get this exception when resizing. I load the page with small size and stretch it:
![web_responsive_exception_1](https://user-images.githubusercontent.com/731270/148581243-b162c353-1415-404c-a46d-8bad7b2c1e13.gif)

Please, update the readme with the explanation of how shortcuts works...

Please apply/adapt this commits: 622991abe3fa56013d0ec36292092f3ebf19d554 and 06269897f130764f2802225f8eda5e02a9cb160b

SplashS commented:
> I get this exception when resizing. I load the page with small size and stretch it

Yes, it's the big problem for me. Exception caused in my code that should rerender kanban one2many as list on resize.
But it's initialization ties with action initialization and I still don't have solution to rerender one2many on the fly.
Will try one more time. Anyway any suggestions welcome.

SplashS committed:
[FIX] make base odoo JS tests working with web_responsive module

SplashS commented:
> I get this exception when resizing. I load the page with small size and stretch it:

I fixed it but kanban/list mode doesn't switch automatically for X2Many fields. Anyway it works.

> Please, update the readme with the explanation of how shortcuts works...

If I understand right mergebot will update README from readme/DESCRIPTION. Here I updated shortcuts explanation.

> Please apply/adapt this commits: [622991a](https://github.com/OCA/web/commit/622991abe3fa56013d0ec36292092f3ebf19d554) and [0626989](https://github.com/OCA/web/commit/06269897f130764f2802225f8eda5e02a9cb160b)

Done

Tardo reviewed (approved):
Thanks for the great work!

pedrobaeza commented:
@SplashS one last functional tests by my part, with a couple of things:

- On the AppSwitcher, if you focus on the search textbox, pressing Esc doesn't close it as before.
- In combination with `web_environment_ribbon`,  the ribbon disappears when hovering on the top bar:

  ![Peek 25-01-2022 14-13](https://user-images.githubusercontent.com/7165771/150983447-a9247826-e474-436d-9911-bea02cb90445.gif)

  Is it a problem of this module or the other?

Finally, I think your company deserves a place in the author list, due to all your work in the latest versions.

SplashS commented:
> * On the AppSwitcher, if you focus on the search textbox, pressing Esc doesn't close it as before.

Yep! Not the simplest one solution, but it's fixed.

> * In combination with `web_environment_ribbon`,  the ribbon disappears when hovering on the top bar:
>   Is it a problem of this module or the other?

I removed `web_responsive` from apps and `web_environment_ribbon` behavior is the same. So I think it's on ribbon module side. Quick look gives me no idea why. But it disappears on hovering navbar in any point.

> Finally, I think your company deserves a place in the author list, due to all your work in the latest versions.

Thank you, I've forgotten about this place.

pedrobaeza commented:
/ocabot merge nobump

jcdrubay commented:
Thank you @SplashS and all who supported for this great migration :)
